### ISSUE TITLE
Error: Cannot read properties of undefined (reading 'v2')

### ISSUE THREAD

**Issue by WarboxLiam** [winteragency/sanity-plugin-link-field#13]
I get this error whenever I go to a page / UI element that contains a link.

I have added the plugin to my sanity.config.ts like so:

```ts
/**
 * This configuration is used to for the Sanity Studio that’s mounted on the `\src\app\admin\[[...index]]\page.tsx` route
 */
import { codeInput } from '@sanity/code-input'
import { table } from '@sanity/table'
import { visionTool } from '@sanity/vision'

import { defineConfig } from 'sanity'
import { linkField } from 'sanity-plugin-link-field'
import { media } from 'sanity-plugin-media'
import { structureTool } from 'sanity/structure'

// Go to https://www.sanity.io/docs/api-versioning to learn how API versioning works
import { apiVersion, dataset, projectId } from './sanity/env'
import { schema } from './sanity/schema'

export default defineConfig({
	basePath: '/admin',
	projectId,
	dataset,
	// Add and edit the content schema in the './sanity/schema' folder
	schema,
	plugins: [
		structureTool(),
		media(),
		// Vision is a tool that lets you query your content with GROQ in the studio
		// Only loaded in development mode
		// https://www.sanity.io/docs/the-vision-plugin
		process.env.NODE_ENV === 'development'
			? visionTool({ defaultApiVersion: apiVersion })
			: { name: 'vision-tool-disabled' },
		// Inputs
		codeInput(),
		table(),
		linkField({
			linkableSchemaTypes: ['page'],
		}),
	],
})
```

This happens if it's in blockContent (which has been added like so;

```ts
annotations: [
    defineField({
        title: 'Link',
        name: 'Link',
        type: 'link',
        icon: InternalLinkIcon,
    }),
],
```

Or whether it's used directly on a document in a regular field, similar the the above.

**Comment by WarboxLiam** [comment]
package.json

```json
{
	"name": "my-project",
	"version": "0.1.0",
	"private": true,
	"scripts": {
		"dev": "next dev",
		"build": "next build",
		"start": "next start",
		"lint": "next lint",
		"lint:fix": "next lint --fix",
		"schema:extract": "npx sanity schema extract --enforce-required-fields",
		"schema:typegen": "npx sanity typegen generate"
	},
	"dependencies": {
		"@portabletext/react": "^3.0.18",
		"@sanity/code-input": "^4.1.4",
		"@sanity/icons": "^3.1.0",
		"@sanity/image-url": "^1.0.2",
		"@sanity/table": "^1.1.2",
		"@sanity/vision": "^3.37.2",
		"@splidejs/react-splide": "^0.7.12",
		"clsx": "^2.1.0",
		"framer-motion": "^11.0.28",
		"next": "14.2.0",
		"next-sanity": "^9.3.10",
		"next-sanity-image": "^6.1.1",
		"react": "^18.3.1",
		"react-dom": "^18.3.1",
		"react-hook-form": "^7.51.3",
		"react-icons": "^5.0.1",
		"sanity": "^3.45.0",
		"sanity-plugin-link-field": "^1.3.0",
		"sanity-plugin-media": "^2.2.5",
		"sharp": "^0.33.3",
		"styled-components": "^6.1.8",
		"tailwind-merge": "^2.2.2"
	},
	"devDependencies": {
		"@svgr/webpack": "^8.1.0",
		"@tailwindcss/container-queries": "^0.1.1",
		"@tailwindcss/typography": "^0.5.13",
		"@trivago/prettier-plugin-sort-imports": "^4.3.0",
		"@types/node": "^18.19.34",
		"@types/react": "^18.3.3",
		"@types/react-dom": "^18.3.0",
		"autoprefixer": "^10.4.19",
		"eslint": "^8.56",
		"eslint-config-next": "14.2.0",
		"eslint-config-prettier": "^9.1.0",
		"postcss": "^8.1",
		"prettier-eslint": "^16.3.0",
		"prettier-plugin-tailwindcss": "^0.6.1",
		"tailwindcss": "^3.4.4",
		"typescript": "^5.4.5"
	}
}

```

**Comment by marcusforsberg** [comment]
Hi @WarboxLiam, thanks for opening this issue.
I'm not able to reproduce this error on my end. Can you please provide a minimum reproduction repo so I can take a look? Thanks!

**Comment by WarboxLiam** [comment]
This had something to do with my pnpm-lock.yaml

I created a new blank project and added the link to the schema and everything work well.

I slowly migrated all of my schema over from the project that had the issue and it still worked.

I then just straight up copied my sanity folder into the new project and again, it worked just fine.

I copied over my package.json and reran `pnpm install` in the new project and STILL everything worked fine.

------------

What got this working was removing my `node_modules` folder, and my `pnpm-lock.yaml` file and doing a fresh install. The link now appears to be working
